optking/optimize.py

```diff
@@ -107,18 +107,66 @@ def __init__(self, molsys: Molsys, history_object: history.History, params: op.O
         self.step_number = 0
         self.computer = computer
         self.linesearch_method = None
+        self.stashed_hessian = None
 
         if params.linesearch:
             self.linesearch_method = OptimizationManager._LINESEARCHES["ENERGY"](molsys, self.history, params)
             self.opt_method.trust_radius_on = False
-            self.stashed_hessian = None
         self.requires = self.update_requirements()
         self.current_requirements = self.update_requirements()
         self.params = params
         self.erase_hessian = False
         self.check_linesearch = True
         self.error = None
 
+    def to_dict(self):
+        """ Convert attributes to serializable form. """
+        d = {
+            "direction": self.direction,
+            "step_number": self.step_number,
+            "stashed_hessian": self.stashed_hessian,
+            "requires": self.requires,
+            "current_requirements": self.current_requirements,
+            "erase_hessian": self.erase_hessian,
+            "check_linesearch": self.check_linesearch,
+            "error": self.error
+        }
+        if self.linesearch_method:
+            d["linesearch_method"] = self.linesearch_method.to_dict()
+
+        if self.params.opt_type == "IRC":
+            d["irc_object"] = self.opt_method.to_dict()
+
+        return d
+
+    @classmethod
+    def from_dict(cls, d, molsys, history, params, computer):
+        """ Reload attributes from the provided dictionary. Create all necessary classes.
+        
+        To prevent duplication, OptHelper handles converting the molsys, history, params, and computer to/from dict
+        """
+
+        manager = cls(molsys, history, params, computer)
+        manager.direction = d["direction"]
+        method = "IRC" if params.opt_type == "IRC" else params.step_type
```

Comment:
In the future, we may want to support method == "IRC" and also step_type = "NR"/"RFO",  etc..  Currently unsupported?

Reply:
Is this to run an optimization directly following an IRC? I'm unclear what the behavior of an IRC + RFO would be.

Reply:
Each point along the IRC is the result of a constrained optimization.  But using the projected forces one could search for the point using line search, NR, or RFO methods, no?  The RFO might be a little weird.  I haven't thought about the IRC algorithm for awhile.

Reply:
I think as implemented, we're using NR.  Implicit in that Eqn. 24 in dq_irc, but I wouldn't swear to it.

Reply:
Anyway, not a front-burner issue, and we can merge at will.